DeusEx/Classes/DXRReduceItems.uc

```diff
@@ -1,20 +1,119 @@
 class DXRReduceItems extends DXRActorsBase;
 
+struct sReduceAmmo {
+    var class<Ammo> type;
+    var int percent;
+};
+struct sReduceItem {
+    var class<Actor> type;
+    var int percent;
+};
+struct sSetMax {
+    var class<DeusExPickup> type;
+    var int percent;
+};
+struct sMaxAmmo {
+    var class<Ammo> type;
+    var int percent;
+};
+
+var config int mission_scaling[16];
+var config sReduceAmmo ammo_reductions[16];
+var config sReduceItem reduce_items[16];
+var config sSetMax max_copies[16];
+var config sMaxAmmo max_ammo[16];
+
+function CheckConfig()
+{
+    local int i;
+    if( config_version == 0 ) {
+        for(i=0; i < ArrayCount(mission_scaling); i++) {
+            mission_scaling[i] = 100;
+        }
+        for(i=0; i < ArrayCount(ammo_reductions); i++) {
+            ammo_reductions[i].type = None;
+        }
+        for(i=0; i < ArrayCount(reduce_items); i++) {
+            reduce_items[i].type = None;
+        }
+        for(i=0; i < ArrayCount(max_copies); i++) {
+            max_copies[i].type = None;
+        }
+        for(i=0; i < ArrayCount(max_ammo); i++) {
+            max_ammo[i].type = None;
+        }
+    }
+    Super.CheckConfig();
+}
+
 function FirstEntry()
 {
+    local int i, mission, scale;
     Super.FirstEntry();
 
-    ReduceAmmo( float(dxr.flags.ammo)/100.0 );
-    ReduceSpawns(class'Multitool', dxr.flags.multitools);
-    ReduceSpawns(class'Lockpick', dxr.flags.lockpicks);
-    ReduceSpawns(class'BioelectricCell', dxr.flags.biocells);
-    ReduceSpawns(class'MedKit', dxr.flags.medkits);
+    mission = Clamp(dxr.dxInfo.missionNumber, 0, ArrayCount(mission_scaling)-1);
+    scale = mission_scaling[mission];
```

Comment:
now if we wanted, we could make mission 1 not reduce how many items the player gets, and make area 51 greatly reduce how many items the player gets